js/ui/notify.js

```diff
@@ -29,4 +39,100 @@ const notify = function(message, /* optional */ type, displayTime) {
+};
+
+const getStackContainer = (id) => {
+    if($(`#${id}`).length) {
+        return $(`#${id}`);
+    } else {
+        return createStackContainer(id);
+    }
+};
+
+
+const setContainerClasses = (container, direction) => {
+    container.removeClass();
+    container.addClass('dx-toast-container').addClass(`dx-toast-container-${direction}-direction`);
+};
+
+const setContainerStyles = (container, direction, position) => {
+    const styles = typeof position === 'string'
+        ? getPositionStylesByAlias(container, direction, position)
+        : getPositionStylesByCoordinates(container, direction, position);
+
+    container.css(styles);
+};
+
+const getPositionStylesByAlias = (container, direction, position) => {
+    const toastWidth = container.children().first()[0].offsetWidth;
+    const toastHeight = container.children().first()[0].offsetHeight;
+    const windowHeight = window.innerHeight;
+    const windowWidth = window.innerWidth;
+
+    switch(position) {
+        case 'top left':
+            return getPositionStylesByCoordinates(container, direction, { top: 10, left: 10 });
+        case 'top right':
+            return getPositionStylesByCoordinates(container, direction, { top: 10, right: 10 });
+        case 'bottom left':
+            return getPositionStylesByCoordinates(container, direction, { bottom: 10, left: 10 });
+        case 'bottom right':
+            return getPositionStylesByCoordinates(container, direction, { bottom: 10, right: 10 });
+        case 'top center':
+            return getPositionStylesByCoordinates(container, direction, { top: 10, left: Math.round(windowWidth / 2 - toastWidth / 2) });
+        case 'left center':
+            return getPositionStylesByCoordinates(container, direction, { top: Math.round(windowHeight / 2 - toastHeight / 2), left: 10 });
+        case 'right center':
+            return getPositionStylesByCoordinates(container, direction, { top: Math.round(windowHeight / 2 - toastHeight / 2), right: 10 });
+        case 'center':
+            return getPositionStylesByCoordinates(container, direction, {
+                top: Math.round(windowHeight / 2 - toastHeight / 2),
+                left: Math.round(windowWidth / 2 - toastWidth / 2)
+            });
+        case 'bottom center':
+        default:
+            return getPositionStylesByCoordinates(container, direction, { bottom: 10, left: Math.round(windowWidth / 2 - toastWidth / 2) });
+    }
+};
+
+const getPositionStylesByCoordinates = (container, direction, position) => {
+    const styles = {};
+    const toastWidth = container.children().first()[0].offsetWidth;
+    const toastHeight = container.children().first()[0].offsetHeight;
+
+    switch(direction.replace('-reverse', '')) {
+        case 'up':
+            styles.bottom = position.bottom ? position.bottom : `calc(100vh - ${toastHeight}px - ${position.top}px)`;
+            styles.top = '';
+            styles.left = position.left ? `${position.left}px` : '';
```

Comment:
let's create separate util `normalizePositionStyle` to get rid of ternary operations